plugin/src/lib.rs

```diff
@@ -473,12 +474,121 @@ impl<'a, 'cx> Folder for MutatorPlugin<'a, 'cx> {
+                                        s,
+                                        &[
+                                            "sub one to int constant",
+                                        ],
+                                    );
+                                }
+
+                                mut_expression = quote_expr!(self.cx,
+                                    {
+                                        if mutagen::now($n) { $l - 1 }
+                                        else { $mut_expression }
+                                    });
+                            }
+
+                            if int_constant_can_add_one(i as u64, ty) {
+                                let n;
+                                {
+                                    n = self.current_count;
+                                    add_mutations(
+                                        &self.cx,
+                                        &mut self.mutations,
+                                        &mut self.current_count,
+                                        s,
+                                        &[
+                                            "add one to int constant",
+                                        ],
+                                    );
+                                }
+
+                                mut_expression = quote_expr!(self.cx,
+                                    {
+                                        if mutagen::now($n) { $l + 1 }
+                                        else { $mut_expression }
+                                    });
+                            }
+
+                            mut_expression
+                        },
+                        _ => {
+                            P(Expr {
+                                id,
+                                node: ExprKind::Lit(P(l)),
+                                span,
+                                attrs,
+                            })
+                        }
+                    }
+                });
+
+                exp
+
+            },
+            e => {
+                P(fold::noop_fold_expr(e, self))
+            },
         }) //TODO: more expr mutations
     }
 }
 
+fn int_constant_can_subtract_one(i: i64, ty: LitIntType) -> bool{
+    let min: i64 = match ty {
+        LitIntType::Unsuffixed | LitIntType::Unsigned(_) => 0,
+        LitIntType::Signed(IntTy::Isize) => std::i32::MIN as i64,
+        LitIntType::Signed(IntTy::I8) => std::i8::MIN as i64,
+        LitIntType::Signed(IntTy::I16) => std::i16::MIN as i64,
+        LitIntType::Signed(IntTy::I32) => std::i32::MIN as i64,
+        LitIntType::Signed(IntTy::I64) => std::i64::MIN as i64,
+        _ => std::i64::MIN,
+    };
+
+    i as i64 > min
+}
+
+fn int_constant_can_add_one(i: u64, ty: LitIntType) -> bool {
+    let max: u64 = match ty {
+        LitIntType::Unsuffixed => 256,
+        LitIntType::Unsigned(UintTy::Usize) => std::u32::MAX as u64,
```

Comment:
I doubt which should be the max value for usize. As far as I know, it's system dependent, so I'm not sure which is the most safe value here.

Reply:
This shouldn't be a problem as long as we're consistent.